src/services/item/repository.ts

```diff
@@ -287,6 +287,39 @@ export class ItemRepository {
     return farthestItem?.path?.split('.')?.length ?? 0;
   }
 
+  /**
+   * Return all the items where the creator is the given actor.
+   * // TODO: check if it is wanted !
+   * It even returns the item if the actor is the creator but without permissions on it !
+   *
+   * @param memberId The creator of the items.
+   * @returns an array of items created by the actor.
+   */
+  async getForMemberExport(memberId: string): Promise<Item[]> {
+    return this.repository
+      .createQueryBuilder('item')
+      .select([
+        'item.id',
+        'item.name',
+        'item.type',
+        'item.description',
+        'item.path',
+        'item.extra',
+        'item.settings',
+        'item.lang',
+        'item.displayName',
```

Comment:
Actually I'm thinking... if we want to add a new column it will be missing in all the select you do. I guess it will also be missing in the schema, but I wonder if we can more easily gather the info 🤔

Reply:
I have written a schema to TypeORM selection Mapper, what do you think ?

Reply:
Don't hesitate to check the code `src/services/member/plugins/data/utils/selection.utils.ts` :)